public/app/plugins/datasource/grafana-azure-monitor-datasource/components/ResourcePicker/ResourcePicker.tsx

```diff
@@ -104,120 +109,161 @@ const ResourcePicker = ({
+        <p className={styles.resultLimit}>Showing first {resourcePickerData.resultLimit} results</p>
       ) : (
-        <>
-          <table className={styles.table}>
-            <thead>
-              <tr className={cx(styles.row, styles.header)}>
-                <td className={styles.cell}>Scope</td>
-                <td className={styles.cell}>Type</td>
-                <td className={styles.cell}>Location</td>
+        <Space v={2} />
+      )}
+
+      <table className={styles.table}>
+        <thead>
+          <tr className={cx(styles.row, styles.header)}>
+            <td className={styles.cell}>Scope</td>
+            <td className={styles.cell}>Type</td>
+            <td className={styles.cell}>Location</td>
+          </tr>
+        </thead>
+      </table>
+
+      <div className={styles.tableScroller}>
+        <table className={styles.table}>
+          <tbody>
+            {isLoading && (
+              <tr className={cx(styles.row)}>
+                <td className={styles.cell}>
+                  <LoadingPlaceholder text={'Loading...'} />
+                </td>
               </tr>
-            </thead>
-          </table>
-
-          <div className={styles.tableScroller}>
-            <table className={styles.table}>
-              <tbody>
-                {rows.map((row) => (
-                  <NestedRow
-                    key={row.uri}
-                    row={row}
-                    selectedRows={selectedResourceRows}
-                    level={0}
-                    requestNestedRows={requestNestedRows}
-                    onRowSelectedChange={handleSelectionChanged}
-                    selectableEntryTypes={selectableEntryTypes}
-                    scrollIntoView={true}
-                  />
-                ))}
-              </tbody>
-            </table>
-          </div>
-
-          <div className={styles.selectionFooter}>
-            {selectedResourceRows.length > 0 && (
-              <>
-                <h5>Selection</h5>
-
-                <div className={styles.tableScroller}>
-                  <table className={styles.table}>
-                    <tbody>
-                      {selectedResourceRows.map((row) => (
-                        <NestedRow
-                          key={row.uri}
-                          row={row}
-                          selectedRows={selectedResourceRows}
-                          level={0}
-                          requestNestedRows={requestNestedRows}
-                          onRowSelectedChange={handleSelectionChanged}
-                          selectableEntryTypes={selectableEntryTypes}
-                        />
-                      ))}
-                    </tbody>
-                  </table>
-                </div>
-                <Space v={2} />
-              </>
             )}
+            {!isLoading && rows.length === 0 && (
+              <tr className={cx(styles.row)}>
+                <td className={styles.cell}>No resources found</td>
```

Comment:
When I use VoiceOver, this doesn't get announced. I think we can add an `aria-live="polite"` here to announce when no results are found, although I'm not actually sure if that's the correct way to do it.

Reply:
Yeah, I'm not entirely sure either but in manually testing it out it seems reasonable to me!